duo_client/admin.py

```diff
@@ -519,6 +527,95 @@ def get_authentication_log(self, api_version=1, **kwargs):
+        To access this endpoint,the customer needs to be enabled by agents manually.
+
+
+        mintime (required) - Unix timestamp in ms; fetch records >= mintime
+        maxtime (required) - Unix timestamp in ms; fetch records <= maxtime
+        limit - Number of results to limit to
+        next_offset - Used to grab the next set of results from a previous response
+        sort - Sort order to be applied
+
+        Returns:
+            {
+                "items": [
+                    {
+                        "access_device": {
+                            "browser": <str:browser>,
+                            "browser_version": <str:browser version>,
+                            "ip": {
+                                "address":  <str:ip address>
+                            },
+                            "location": {
+                                "city":  <str:city>,
+                                "country":  <str:country>,
+                                "state": <str:state>
+                            },
+                            "os":  <str:os name>,
+                            "os_version":  <str:os_version>
+                        },
+                        "action":  <str:action>,
+                        "activity_id":  <str:activity id>,
+                        "actor": {
+                            "details": <str:json for actor details>
+                            "key" :  <str:actor's key>,
+                            "name":  <str:actor's name>,
+                            "type":  <str:actor type>
+                        },
+                        "akey":  <str:akey>,
+                        "application": {
+                            "key":  <str:application's key>,
+                            "name":  <str:application's name>,
+                            "type":  <str:application's type>
+                        },
+                        "target": {
+                            "details":  <str:target's details if available> ,
+                            "key" :  <str:target's key>,
+                            "name":  <str:target's name>,
+                            "type":  <str:target's type>
+                        },
+                        "ts":  <str:timestamp captured for activity>
+                    },
+                ],
+                "metadata": {
+                    "next_offset": [
+                        "1532951895000",
+                        "af0ba235-0b33-23c8-bc23-a31aa0231de8"
+                    ],
+                    "total_objects": 1
+                }
+            },
+
+        Raises RuntimeError on error.
+        """
+        params = {}
+
+        for k in kwargs:
+            if kwargs[k] is not None and k in VALID_ACTIVITY_REQUEST_PARAMS:
+                params[k] = kwargs[k]
+
+        if 'mintime' in params:
+            params['mintime'] = str(int(params['mintime']))
```

Comment:
As far as I can tell, on the Duo side, we're expecting mintime and maxtime to be integers?  Above (lines 505 and 511) we don't convert back to strings for mintime and maxtime in the authlog method.

Reply:
I actually referred this - https://github.com/duosecurity/duo_client_python/blob/master/duo_client/admin.py#L326
Let me know if I need to correct it...

Reply:
I think these get x-www-form-urlencoded so an integer is passed as a string that happens to consist only as digits? If I remember correct (and I might not) normalize_params() and canon_params() require every value to be a string (or list of strings) already.

`'{:d}'.format(int(params['mintime']))` on line 505 is a way of converting it to a string that makes sure the input is int-ish, unlike str() which accepts most any old thing. This extra safety makes little difference here because the input to str() is the return value of int().